diff --git a/node_modules/vue-tsc/index.js b/node_modules/vue-tsc/index.js
index a680d25..eb49461 100644
--- a/node_modules/vue-tsc/index.js
+++ b/node_modules/vue-tsc/index.js
@@ -4,7 +4,8 @@ exports.run = run;
 const runTsc_1 = require("@volar/typescript/lib/quickstart/runTsc");
 const vue = require("@vue/language-core");
 const windowsPathReg = /\\/g;
-function run(tscPath = require.resolve('typescript/lib/tsc')) {
+const semver = require('semver');
+function run(tscPath = getTscPath()) {
     let runExtensions = ['.vue'];
     const extensionsChangedException = new Error('extensions changed');
     const main = () => (0, runTsc_1.runTsc)(tscPath, runExtensions, (ts, options) => {
@@ -35,4 +36,16 @@ function run(tscPath = require.resolve('typescript/lib/tsc')) {
         }
     }
 }
+
+function getTscPath() {
+	const version = require('typescript/package.json').version;
+
+	if (semver.gte(version, '5.7.0')) {
+		return require.resolve('typescript/lib/_tsc');
+	}
+	else {
+		return require.resolve('typescript/lib/tsc');
+	}
+}
+
 //# sourceMappingURL=index.js.map
